### PR TITLE
Push build number to ecr as tag together with latest

### PR DESCRIPTION
**Pull request by anguss00**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -57,7 +57,9 @@ pipeline {
                         docker.build("javaduck:${env.BUILD_ID}", "aggregation-worker/")
                         withEnv(['PATH+EXTRA=/var/lib/jenkins/bin']) {
                             docker.withRegistry(env.ECR_REGISTRY_URL) {
-                                docker.image("javaduck:${env.BUILD_ID}").push("latest")
+                                dockerImage = docker.image("javaduck:${env.BUILD_ID}")
```
Comment (lwgordonimos) [inline Jenkinsfile:60]: Should probably just use BUILD_NUMBER here for consistency: https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables

>BUILD_ID
    The current build ID, identical to BUILD_NUMBER for builds created in Jenkins versions 1.597+

The doco here: https://www.jenkins.io/doc/book/pipeline/docker/ has an example where push is run with no parameters. I'm wondering if we should just be doing something simpler, e.g.

```groovy
dockerImage = docker.image("javaduck:${env.BUILD_ID}")
dockerImage.push()
dockerImage.push("latest")
```

It seems we're building with one local name, pushing to a different remote name and then tagging latest.

Comment (lwgordonimos) [reply Jenkinsfile:60]: Or if it doesn't work as expected, or gets overcomplicated, we can just run with it, but changingBUILD_ID to BUILD_NUMBER at least for clarity that they are actually the same thing.

Comment (lwgordonimos) [inline Jenkinsfile:60]:
```suggestion
                                dockerImage = docker.image("javaduck:${env.BUILD_NUMBER}")
```

File: Jenkinsfile [hunk 2]
```diff
@@ -53,15 +53,17 @@ pipeline {
             }
             steps {
                 script {
-                    if (env.BRANCH_NAME == 'master') {
+                    if (env.BRANCH_NAME == 'aws_wps_ecr_pipeline') {
```
Comment (lwgordonimos) [inline Jenkinsfile:56]: `aws_wps_ecr_pipeline` is the new master?

Comment (anguss00) [reply Jenkinsfile:56]: Sharp, testing off this branch ;-)

File: Jenkinsfile [hunk 3]
```diff
@@ -54,14 +54,16 @@ pipeline {
...
                             docker.withRegistry(env.ECR_REGISTRY_URL) {
-                                docker.image("javaduck:${env.BUILD_ID}").push("latest")
+                                dockerImage = docker.image("javaduck:${env.BUILD_TAG}")
+                                dockerImage.push("${env.BUILD_TAG}")
+                                dockerImage.push("latest")
                             }
                         }
                     } else {
-                        docker.build("javaduck-${env.BRANCH_NAME}:${env.BUILD_ID}", "aggregation-worker/")
+                        docker.build("javaduck-${env.BRANCH_NAME}:${env.BUILD_NUMBER}", "aggregation-worker/")
```
Comment (lwgordonimos) [inline Jenkinsfile:66]: Ok this is interesting now. This is now so close to the BUILD_TAG that I wonder if we shouldn't just be using the actual BUILD_TAG value.

**Pull request merged**